test/rules/no-mocha-global-arrows.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+var RuleTester = require('eslint').RuleTester,
+    rules = require('../../').rules,
+    ruleTester = new RuleTester(),
+    expectedErrorMessage = 'Do not pass arrow functions to it()';
+
+ruleTester.run('no-mocha-global-arrows', rules['no-mocha-global-arrows'], {
+
+    valid: [
+        'it()',
+        'it(function() { assert(something, false); })',
+        'it("should be false", function() { assert(something, false); })',
+        {
+            // In this example, `it` is not a global.
```

Comment:
I don't have a need for it, but it seems like a nice idea.

Doing it well would require more complex logic. First, you'd need to look for `require('mocha')`. Then you'd need to see `var darkBrown = require('mocha')` and then check `darkBrown.anything`. And see `var b4 = require('mocha').beforeEach` and check `b4`.

If you think we should support it in the future, I'd be happy to adjust the interface. Instead of `no-mocha-global-arrows`, it could be `no-mocha-arrows`. Initially it would just check globals, as this does, but if someone wanted to implement the require syntax, they could add a `'globals': true` option which preserves this behavior, and a `'globals': false, 'require': true` option to turn on the new logic. (`'require': true` might not be needed. The only reason I would include it is if it slows down the checking.)

It would share a few lines of code and `mochaFunctionNames` with this rule; otherwise I'd suggest using a separate rule.